src/Compilers/Core/Portable/EmbeddedText.cs

```
@@ -0,0 +1,345 @@
+
+                return builder.ToImmutableArrayAndFree();
+            }
+            else
+            {
+                Debug.Assert(length > 0);
+                var builder = BlobBuildingStream.GetInstance();
+                builder.WriteInt32(length);
+
+                using (var deflater = new DeflateStream(builder, CompressionLevel.Optimal))
+                {
+                    stream.CopyTo(deflater);
+                }
+
+                return builder.ToImmutableArrayAndFree();
+            }
+        }
+
+        internal static ImmutableArray<byte> CreateBlob(byte[] buffer, int length)
+        {
+            Debug.Assert(buffer != null);
+            Debug.Assert(length > 0);
+            Debug.Assert(length <= buffer.Length);
+
+            if (length < CompressionThreshold)
+            {
+                var builder = Cci.PooledBlobBuilder.GetInstance();
+                builder.WriteInt32(0);
+                builder.WriteBytes(buffer, 0, length);
+                return builder.ToImmutableArrayAndFree();
+            }
+            else
+            {
+                Debug.Assert(length > 0);
+                var builder = BlobBuildingStream.GetInstance();
+                builder.WriteInt32(length);
+
+                using (var deflater = new DeflateStream(builder, CompressionLevel.Optimal))
+                {
+                    deflater.Write(buffer, 0, length);
+                }
+
+                return builder.ToImmutableArrayAndFree();
+            }
+        }
+
+        private static ImmutableArray<byte> CreateBlob(SourceText sourceText)
+        {
+            Debug.Assert(sourceText != null);
+            Debug.Assert(sourceText.CanBeEmbedded);
+            Debug.Assert(sourceText.Encoding != null);
+            Debug.Assert(sourceText.PrecomputedEmbeddedTextBlob.IsDefault);
+
+            int maxByteCount = int.MaxValue;
+            try { maxByteCount = sourceText.Encoding.GetMaxByteCount(sourceText.Length);  }
```

Also add a comment why we are swallowing the exception.

---
In reply to: [72831841](https://github.com/dotnet/roslyn/pull/12353#discussion_r72831841) [](ancestors = 72831841)